src/seedu/addressbook/data/AddressBook.java

```diff
@@ -25,12 +27,15 @@
     private final UniquePersonList allPersons;
     private final UniqueTagList allTags; // can contain tags not attached to any person
 
+    public ArrayList<Tagging> listOfTaggingRecords;
+
     /**
      * Creates an empty address book.
      */
     public AddressBook() {
         allPersons = new UniquePersonList();
         allTags = new UniqueTagList();
+        this.listOfTaggingRecords = new ArrayList<>();
```

Comment:
I'm not sure if this can be compiled. If it can, it's still better to define it more explicitly `<Tagging>`.